src/server/detail/wrapped_json_path.h

```diff
@@ -14,16 +14,66 @@
 
 namespace dfly {
 
+class Nothing {};
 using JsonExpression = jsoncons::jsonpath::jsonpath_expression<JsonType>;
 
 template <typename T>
 using JsonPathEvaluateCallback = absl::FunctionRef<T(std::string_view, const JsonType&)>;
 
-template <typename T> class JsonCallbackResult {
+template <typename T = Nothing> class MutateCallbackResult {
  public:
+  MutateCallbackResult() = default;
+
+  explicit MutateCallbackResult(bool should_be_deleted_) : should_be_deleted(should_be_deleted_) {
+  }
+
+  MutateCallbackResult(bool should_be_deleted_, T&& value)
+      : should_be_deleted(should_be_deleted_), value_(std::forward<T>(value)) {
+  }
+
+  bool HasValue() const {
+    return value_.has_value();
+  }
+
+  T&& GetValue() && {
+    return std::move(value_).value();
+  }
+
+  bool should_be_deleted;
+
+ private:
+  std::optional<T> value_;
+};
+
+template <typename T>
+using JsonPathMutateCallback =
+    absl::FunctionRef<MutateCallbackResult<T>(std::optional<std::string_view>, JsonType*)>;
+
+namespace details {
+
+template <typename T> void OptionalEmplace(T&& value, std::optional<T>* optional) {
+  optional->emplace(std::forward<T>(value));
+}
+
+template <typename T>
+void OptionalEmplace(std::optional<T>&& value, std::optional<std::optional<T>>* optional) {
+  if (value.has_value()) {
+    optional->emplace(std::forward<std::optional<T>>(value));
+  }
+}
+
+}  // namespace details
+
+template <typename T, bool SaveFirstResult = false> class JsonCallbackResult {
```

Comment:
if you use = false somewhere, all your types below will break, so it can be just a regular variable or a const (I see it's never changed)